app/dao/fact_notification_status_dao.py

```diff
@@ -242,6 +243,63 @@ def get_last_send_for_api_key(api_key_id):
     ).all()
 
 
+def get_api_key_ranked_by_notifications_created(n_days_back):
+    """
+    SELECT
+        a.api_key_id,
+        a.service_id,
+        a.last_send,
+        a.total_notifications,
+        api_keys.name,
+        api_keys.key_type,
+        services.name
+    FROM (
+        SELECT
+            api_key_id,
+            service_id,
+            max(created_at) as last_send,
+            count(*) as total_notifications
+        FROM notifications
+        WHERE created_at > start_date and api_key_id is not null
+        GROUP BY api_key_id, service_id
+        ORDER BY count(*) DESC
+        LIMIT 50
+    ) as A
+    JOIN api_keys on api_keys.id = A.api_key_id
+    JOIN services on services.id = A.service_id;
+    """
+
+    start_date = datetime.utcnow() - timedelta(days=n_days_back)
+    table_a = db.session.query(
+        Notification.api_key_id,
+        Notification.service_id,
+        func.max(Notification.created_at).label('last_notification_created'),
+        func.count(Notification.id).label('total_notifications')
+    ).filter(
+        Notification.created_at >= start_date,
+        Notification.api_key_id is not None
+    ).group_by(
+        Notification.api_key_id,
+        Notification.service_id
+    ).order_by(
+        func.count(Notification.id).desc()
+    ).subquery()
+
+    return db.session.query(
+        table_a.c.api_key_id,
+        table_a.c.service_id,
+        table_a.c.last_notification_created,
+        table_a.c.total_notifications,
```

Comment:
Not sure what the `.c.` means, I just copied it from another example of a subquery on this page. And it seems to work!